src/models/history/tree-manager.ts

```diff
@@ -77,64 +95,68 @@ export const TreeManager = types
 
   findActiveHistoryEntry(historyEntryId: string) {
     return self.activeHistoryEntries.find(entry => entry.id === historyEntryId);
-  }
-}))
-.actions(self => {
-  let firestoreHistoryInfoPromise: Promise<IFirestoreHistoryInfo> | undefined;
-
-  async function prepareFirestoreHistoryInfo(): Promise<IFirestoreHistoryInfo> {
-    const validateCommentableDocument = 
-      getFirebaseFunction<ICommentableDocumentParams>("validateCommentableDocument_v1");
+  },
 
-    // We'll need provide this context to the tree manager either through a MST env or volatile prop
-    const {userContext, mainDocument, firestore} = self;
-    if (!userContext || !mainDocument || !firestore || !userContext.uid) {
-      console.error("cannot record history entry because environment is not valid", 
-        {userContext, mainDocument, firestore});
-      throw new Error("cannot record history entry because environment is not valid");
+  get historyStatus() : HistoryStatus {
+    if (self.loadingError) {
+      return HistoryStatus.HISTORY_ERROR;
     }
 
-    const networkDocKey = networkDocumentKey(userContext.uid, mainDocument.key, userContext.network);
-    const documentPath = `documents/${networkDocKey}`;
-    const documentRef = firestore.doc(documentPath);
-    const docSnapshot = await documentRef.get();
-    // create a document if necessary
-    if (!docSnapshot.exists) {
-      // FIXME-HISTORY: rename this function to validateFirestoreDocumentMetadata_v1
-      await validateCommentableDocument({context: userContext, document: mainDocument.metadata});
+    const historyLength = self.document.history.length;
+    const {currentHistoryIndex} = self;
+    if (currentHistoryIndex === undefined) {
+      // We should be waiting for the query to figure out the last history entry.
+      return HistoryStatus.FINDING_HISTORY_LENGTH;
+    } else {
+      if (historyLength === 0 && currentHistoryIndex === 0) {
+        // Technically we are not loading the history at this point
+        // But we still don't want to show the playback controls
+        // TODO: change `treeManager.isLoading` to be something more descriptive 
+        return HistoryStatus.NO_HISTORY;
+      } else {
+        if (historyLength >= currentHistoryIndex) {
+          // FIXME: how do we address this????
+          // treeManager.setIsLoading(false);
+          // No message because the playback controls will be rendered instead
+          // setSuccessMessage(null); 
```

Comment:
Are these commented out lines worth keeping around?

Reply:
No they aren't worth keeping. I've removed some other comments in this block that seemed unnecessary.